core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/common/WorkflowActor.scala

```diff
@@ -37,7 +37,7 @@ abstract class WorkflowActor(
   })
 
   val networkCommunicationActor: NetworkSenderActorRef = NetworkSenderActorRef(
-    context.actorOf(NetworkCommunicationActor.props(parentNetworkCommunicationActorRef))
+    context.actorOf(NetworkCommunicationActor.props(parentNetworkCommunicationActorRef, logger))
```

Comment:
I have concern that two actors/threads sharing the same logger instance. Is the logger thread-safe? Would the logging become synchronized operations?

Reply:
We're using scala logging inside WorkflowLogger. I think it's thread-safe.